internal/services/oracledatabase/exadata_infrastructure_resource.go

```diff
@@ -0,0 +1,387 @@
+	StorageCount int64  `tfschema:"storage_count"`
+
+	// Optional
+	CustomerContacts  []string                 `tfschema:"customer_contacts"`
+	MaintenanceWindow []MaintenanceWindowModel `tfschema:"maintenance_window"`
+}
+
+func (ExadataInfraResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		// Azure
+		"location": commonschema.Location(),
+
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		// Required
+		"compute_count": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ValidateFunc: validate.ComputeCount,
+		},
+
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+
+		"shape": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+
+		"storage_count": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ValidateFunc: validate.StorageCount,
+		},
+
+		// Optional
+		"customer_contacts": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"maintenance_window": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"custom_action_timeout_in_mins": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Computed:     true,
+						ValidateFunc: validate.CustomActionTimeoutInMins,
+					},
+
+					"days_of_week": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+
+					"hours_of_day": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeInt,
+						},
+					},
+
+					"is_custom_action_timeout_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Computed: true,
+					},
+
+					"is_monthly_patching_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Computed: true,
+					},
+
+					"lead_time_in_weeks": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Computed:     true,
+						ValidateFunc: validate.LeadTimeInWeeks,
+					},
+
+					"months": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+
+					"patching_mode": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						Computed: true,
+					},
+
+					"preference": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						Computed: true,
+					},
+
+					"weeks_of_month": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeInt,
```

Comment:
this should be validated between 1-5?

Reply:
Internally we want 1-4, I've updated to validate that, thank you.